### PR TITLE
[CI] use sklearnex azure pipelines template for sklearnex CI step

### PR DESCRIPTION
**Pull request by icfaust**
## Description

oneDAL should refer to sklearnex's testing rather than storing scripts on its own locally.  This is a maintenance burden that we can remove.  No performance benchmarks are necessary.  This PR has been enabled by uxlfoundation/scikit-learn-intelex#2125 in order to use the PR-generated GNUMKL build.

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Review by homksei — APPROVED** [review]

File: .ci/env/apt.sh [hunk 1]
```diff
@@ -126,6 +134,6 @@ elif [ "${component}" == "build-sysroot" ] ; then
     build_sysroot "$2" "$3" "$4" "$5"
 else
     echo "Usage:"
-    echo "   $0 [dpcpp|mkl|gnu-cross-compilers|clang-format|dev-base|qemu-apt|qemu-deb|llvm-version|build-sysroot]"
+    echo "   $0 [dpcpp|mkl|tbb|gnu-cross-compilers|clang-format|dev-base|qemu-apt|qemu-deb|llvm-version|build-sysroot]"
```
Comment (Alexandr-Solovev) [inline .ci/env/apt.sh:137]: just based on the fact that you introduced install tbb before, may be its better to have tbb|mkl

File: .ci/env/apt.sh [hunk 2]
```diff
@@ -34,8 +34,13 @@ function install_dpcpp {
...
 }
 
+function install_tbb {
+    sudo apt-get install -y intel-oneapi-tbb-devel-2022.0
+}
+
 function install_mkl {
-    sudo apt-get install -y intel-oneapi-mkl-devel-2025.0 intel-oneapi-tbb-devel-2022.0
+    sudo apt-get install -y intel-oneapi-mkl-devel-2025.0
+    install_tbb
```
Comment (Alexandr-Solovev) [inline .ci/env/apt.sh:43]: should it be called here? May be its better to call both functions? And why we must include it in MKL?

Comment (icfaust) [reply .ci/env/apt.sh:43]: Good question @Alexandr-Solovev, why tbb install was included in mkl I do not know (included in the 2025.0 update I think), but I called one function in the other so that we only have to change the tbb version in one location going forward. Let me know if I should change it.

**Review by Alexandr-Solovev — APPROVED** [review]
